### DISCUSSION TITLE
vscode扩展放进去后webview不能正常显示

### DISCUSSION THREAD

**Discussion by tanjiancheng** [opensumi/core#3315] [General]
**描述你的问题（Describe the bug）**
我自己修改编写的vscode插件放进去extensions后，图标和相关的webview都没有显示

![image](https://github.com/opensumi/core/assets/7401570/e62d724d-3d99-420f-ae16-bdee9b4e366f)
![image](https://github.com/opensumi/core/assets/7401570/c901c282-40f4-4ae3-b441-12a08b64159f)

打开控制台显示
![image](https://github.com/opensumi/core/assets/7401570/1749034f-f03a-466e-8cc5-8e4d23426fa4)

对比同样的插件，我使用theia是正常的
![image](https://github.com/opensumi/core/assets/7401570/59b3baee-5c96-4948-9c19-50cb27f50e8a)
![image](https://github.com/opensumi/core/assets/7401570/4a83a09a-4d12-4ce3-99d3-44861cc28e41)

所以这块是有什么兼容性问题吗？

**预期表现（Expected behavior）**
能正常显示图标和相关的view内容

**环境信息（Environment）**

- OS: Windows10
- Browser: chrome
- OpenSumi Version: 2.26.8

**Comment by bytemain** [comment]
我们的 webview 是基于 iframe 做的，因为 iframe 的一些通信的限制，我们需要通过加载 /webview/index.html 来包裹一层 webview，你这里的报错看起来是加载不到这个地址。

可以参考下这个 webpack 脚本，构建一下 webview 到某个目录，然后让 server 能通过某个路径能访问到这个 webview 的地址：

- https://github.com/opensumi/ide-startup/blob/main/configs/webpack.webview.config.js

webview 的地址可以通过前端 ClientApp 的 webviewEndpoint 属性配置，参考：

- https://github.com/opensumi/ide-startup/blob/247cc60f54fa4b6c09680554c9c7a36530c15701/src/browser/render-app.ts#L29
